### ISSUE TITLE
Warn if `>` is passed without quotes

### ISSUE THREAD

**Issue by JakeSiFive** [nushell/nushell#11197]
### Related problem

I use both nushell and bash and sometimes I forget which shell I'm in. I'll fall back to typing `foo > file` instead of `foo out> file` which generally fails because the program I'm invoking doesn't like `>` as a command line argument. I just had a case however where that argument was ignored and thought I was updating files that I was in fact doing nothing with. The result was still unexpected but it took me quite a while to figure out. My first thought was that I need to redirect stderr not stdout because despite redirecting I could still see the output (a common occurence with stderr).

### Describe the solution you'd like

I think if a user uses `>` as a command line option and doesn't quote it they're probably making a mistake since that's just so uncommon due to the last unteenth years of it meaning something else in more widely used shells.

### Describe alternatives you've considered

If you wanted to you could be really extreme and reserve it perhaps so that it has to be quoted. Or you could be a bit more lax and only warn after the program finishes and only if it returns an exit code of zero

### Additional context and details

Basically, its not always apparent (but 99% time time it is, this is just that 1% remaining case) that I'm mis-using `>` and I'd like a hint in the future about this.

**Comment by JakeSiFive** [comment]
My proposal for that is specifically when an unqoted `>` is passed as a string to a function/process. So In the above you'd get a warning but if you used `get ">" file` you wouldn't
